generators/evaluate/GenerateFirOperationsMap.kt

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.generators.evaluate
+
+import org.jetbrains.kotlin.generators.util.GeneratorsFileUtil
+import org.jetbrains.kotlin.types.KotlinType
+import org.jetbrains.kotlin.utils.Printer
+import java.io.File
+
+val FIR_DEST_FILE: File = File("compiler/fir/evaluate/src/org/jetbrains/kotlin/fir/evaluate/FirOperationsMapGenerated.kt")
```

Comment:
Is the only difference between the 2 generators the destination file and excluded functions and types? Is it possible to refactor to reduce duplicated code?

Reply:
Not really. Types that represent constant kinds are different. Therefore, components of operation keys are also different. In `fir`, as a first step, only numeric types are handled here, and to reduce unchecked cast suppression, `Number` type is used explicitly as many as possible. Since `FirConstKind` is representing only literal kinds, not sure we would end up with really similar generators. But, as I mentioned in the PR description, such kind includes `char` and `string` as well, and extending the evaluator to those types may require many changes as well. Maybe we could think of refactoring at that point.